### PR TITLE
Create physical constants submodule

### PR DESCRIPTION
**Pull request by MilesCranmer**
PR for #26.

1. This creates a `Constants` submodule. 
2. `Units` and `Constants` are now both exported by DynamicQuantities (unsure about this, but feels right)
3. `Constants` is made available within the `@u_str` macro for ease of access. (Some constants are treated like units)

For example:

```julia
julia> mass_energy_of_sun = 1.0 * u"Const.M_sun * Const.c^2"
1.7870936681139022e47 m² kg s⁻²

julia> q1 = 1.5u"C"; q2 = 1e-5u"C";

julia> F = Const.k_e * q1 * q2 / (1u"m")
134813.2768839176 m² kg s⁻²

julia> f = 500u"MHz"
5.0e8 s⁻¹

julia> (Constants.h * f) / Constants.eV
2.0678338484619293e-6
```

The following constants are included.

- `c`: Speed of light in a vacuum. Standard.
- `h`: Planck constant. Standard.
- `hbar`: Reduced Planck constant (h/2π). Standard.
- `e`: Elementary charge. Standard.
- `k_B`: Boltzmann constant. Standard.
- `N_A`: Avogadro constant. Standard.
- `eV`: Electron volt. Standard.
- `R`: Molar gas constant. Standard.
- `F`: Faraday constant. Standard.
- `sigma_sb`: Stefan-Boltzmann constant. Standard.
- `alpha`: Fine-structure constant. Measured.
- `u`: Atomic mass unit (1/12th the mass of Carbon-12). Measured.
- `G`: Newtonian constant of gravitation. Measured.
- `mu_0`: Vacuum magnetic permeability. Measured.
- `eps_0`: Vacuum electric permittivity. Measured.
- `m_e`: Electron mass. Measured.
- `m_p`: Proton mass. Measured.
- `m_n`: Neutron mass. Measured.
- `a_0`: Bohr radius. Measured.
- `k_e`: Coulomb constant (Note: SI units only!). Measured.
- `Ryd`: Rydberg frequency. Measured.

I included the most common astro ones too (It might be nice if someone added similar sets of very common constants from other fields.)

- `M_earth`: Earth mass. Measured.
- `M_sun`: Solar mass. Measured.
- `M_jup`: Jupiter mass. Measured.
- `R_earth`: Nominal Earth equatorial radius. Standard.
- `R_jup`: Nominal Jupiter equatorial radius. Standard.
- `R_sun`: Nominal solar radius. Standard.
- `L_sun`: Nominal solar luminosity. Standard.
- `L_bol0`: Standard luminosity at absolute bolometric magnitude 0. Standard.
- `sigma_T`: Thomson scattering cross-section. Measured.
- `au`: Astronomical unit. Standard.
- `pc`: Parsec. Standard.
- `ly`: Light year. Standard.
- `atm`: Standard atmosphere. Standard.

"Standard" means it is an exact standard, while "Measured" means there is some uncertainty associated with it. 

cc @odow @ChrisRackauckas. I am eager to hear your take on this interface given your interest in using the package downstream.

### CODE REVIEW

File: src/constants.jl [hunk 1]
```diff
@@ -0,0 +1,89 @@
+module Const
```
Comment (odow) [inline src/constants.jl:1]:
```suggestion
module Constants
```

There's no need for abbreviations?

File: src/units.jl [hunk 2]
```diff
@@ -145,4 +145,7 @@ macro u_str(s)
     return esc(uparse(s))
 end
 
+include("constants.jl")
+import .Const
```
Comment (MilesCranmer) [inline src/units.jl:149]:
```suggestion
import .Constants
```

File: src/DynamicQuantities.jl [hunk 3]
```diff
@@ -8,10 +9,10 @@ include("fixed_rational.jl")
 include("types.jl")
 include("utils.jl")
 include("math.jl")
-include("units.jl")
+include("units.jl") # < include("constants.jl")
 
 import Requires: @init, @require
-import .Units: uparse, @u_str
+import .Units: uparse, @u_str, Const
```
Comment (MilesCranmer) [inline src/DynamicQuantities.jl:15]:
```suggestion
import .Units: uparse, @u_str, Constants
```

File: src/DynamicQuantities.jl [hunk 4]
```diff
@@ -1,5 +1,6 @@
 module DynamicQuantities
 
+export Units, Const
```
Comment (MilesCranmer) [inline src/DynamicQuantities.jl:3]:
```suggestion
export Units, Constants
```

**Comment by MilesCranmer** [comment]
Combining with verbatim units branch, so closing.

**Comment by MilesCranmer** [comment]
Continued on #32. This PR also introduces the symbolic dimensions to help with https://github.com/trulsf/UnitJuMP.jl/issues/18

**Pull request closed without merging**